src/frontend/js/utils/api/web-analytics/google_tag_manager.ts

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable class-methods-use-this */
+import { BaseWebAnalyticsApi } from './base';
+
+/**
+ *
+ * Google Tag Manager Richie Web Analytics API Implementation
+ *
+ * This implementation is used when web analytics is configured has `google_tag_manager`.
+ * It will send events to the google tag manager.
+ *
+ */
+export default class GoogleTagManagerApi extends BaseWebAnalyticsApi {
+  constructor() {
+    super();
+    if (window.dataLayer === undefined) {
+      throw new Error('Incorrect configuration on Google Tag Manager on Web Analytics.');
+    }
+  }
+  sendEvent(category: string, action: string, label: string, value?: number): void {
+    window.dataLayer?.push({
```

Comment:
As you did in google_analytics : you could store window.dataLayer into a class attribute. In this way, you do not need to check if dataLayer is defined every time you uses it.
```suggestion
    this.dataLayer.push({
```

Reply:
Done. Changed to:

```ts
/* eslint-disable class-methods-use-this */
import { BaseWebAnalyticsApi } from './base';

/**
 *
 * Google Tag Manager Richie Web Analytics API Implementation
 *
 * This implementation is used when web analytics is configured has `google_tag_manager`.
 * It will send events to the google tag manager.
 *
 */
export default class GoogleTagManagerApi extends BaseWebAnalyticsApi {
  dataLayer: Exclude<typeof window.dataLayer, undefined>;

  constructor() {
    super();
    if (window.dataLayer === undefined) {
      throw new Error('Incorrect configuration on Google Tag Manager on Web Analytics.');
    }
    this.dataLayer = window.dataLayer;
  }
  sendEvent(category: string, action: string, label: string, value?: number): void {
    this.dataLayer.push({
      event: 'courseEnroll',
      eventProps: {
        category,
        action,
        label,
        value,
      },
    });
  }
}
```